syntax = "proto3";

package dev.local;

message NestedObj{
    int32 age = 1;
    string first_name = 2;
    string last_name = 3;
    bytes small_image = 4;
    bool is_verified = 5;
    float height = 6;

    repeated string phone_numbers = 7;


    enum EyeColor{
        UNKNOWN = 0;
        BLACK = 1;
        BROWN = 2;
        GREEN = 3;
        BLUE = 4;
    }
    EyeColor eye_color = 8;

    message Date{
        int32 year = 1;
        int32 month = 2;
        int32 day = 3;
    }
    Date birth_date = 9;

    message Address{
        string address_line_1 = 1;
        string address_line_2 = 2;
        string zip_code = 3;
        string city = 4;
        string country = 5;
    }
    repeated Address addresses = 10;
}